addons/account/tests/test_reconciliation.py

```diff
@@ -16,47 +15,36 @@ class TestReconciliationExec(TestAccountReconciliationCommon):
 
     def test_statement_usd_invoice_eur_transaction_eur(self):
         customer_move_lines, supplier_move_lines = self.make_customer_and_supplier_flows(self.currency_euro_id, 30, self.bank_journal_usd, 42, 30, self.currency_euro_id)
-        self.assertRecordValues(customer_move_lines, [
-            {'debit': 0.0,      'credit': 30.0,     'amount_currency': -42, 'currency_id': self.currency_usd_id},
+        self.assertRecordValues(customer_move_lines.sorted('balance', reverse=True), [
             {'debit': 30.0,     'credit': 0.0,      'amount_currency': 42,  'currency_id': self.currency_usd_id},
+            {'debit': 0.0,      'credit': 30.0,     'amount_currency': 0.0, 'currency_id': False},
         ])
-        self.assertRecordValues(supplier_move_lines, [
-            {'debit': 30.0,     'credit': 0.0,      'amount_currency': 42,  'currency_id': self.currency_usd_id},
+        self.assertRecordValues(supplier_move_lines.sorted('balance', reverse=True), [
+            {'debit': 30.0,     'credit': 0.0,      'amount_currency': 0.0, 'currency_id': False},
```

Comment:
is the 0.0 amount_currency a side effect of the task? it's probably a good thing to do but I don't remember we discussed that point

Reply:
We discussed this point a long time ago at the beginning of this task. The only think I remember is I changed something on `account_payment` to make this behavior.

Reply:
to be double check....

Reply:
This behavior is because there is an explicit foreign currency set on the statement line that was the same of the invoice one.
I changed this behavior because if you say explicitly your transaction is in EUR even your bank journal is USD, you expect to match an invoice/payment in USD, not EUR.